src/index.js

```diff
@@ -62,79 +129,108 @@ class LumigoPlugin {
 		}
 
 		const token = _.get(this.serverless.service, "custom.lumigo.token");
-		const pinVersion = _.get(this.serverless.service, "custom.lumigo.pinVersion");
-		const skipInstallNodeTracer = _.get(
-			this.serverless.service,
-			"custom.lumigo.skipInstallNodeTracer",
-			false
-		);
-		let skipReqCheck = _.get(
-			this.serverless.service,
-			"custom.lumigo.skipReqCheck",
-			false
-		);
-
-		let parameters = _.get(this.serverless.service, "custom.lumigo", {});
-		parameters = _.omit(parameters, [
-			"pinVersion",
-			"skipReqCheck",
-			"skipInstallNodeTracer"
-		]);
-		if (token === undefined) {
+		if (!token) {
 			throw new this.serverless.classes.Error(
 				"serverless-lumigo: Unable to find token. Please follow https://github.com/lumigo-io/serverless-lumigo"
 			);
 		}
 
-		if (runtime === "nodejs") {
-			if (!skipInstallNodeTracer) {
-				await this.installLumigoNodejs(pinVersion);
-			}
-
+		if (this.useLayers) {
 			for (const func of functions) {
-				const handler = await this.createWrappedNodejsFunction(
-					func,
-					token,
-					parameters
-				);
+				const funcRuntime = func.runtime || runtime;
+				func.layers = func.layers || [];
+				const layer = await this.getLayerArn(funcRuntime);
+				func.layers.push(layer);
+				func.environment = func.environment || {};
+				func.environment["LUMIGO_ORIGINAL_HANDLER"] = func.handler;
```

Comment:
I'm missing here also the token environment variable

Reply:
good catch